lib/data/external/GCP/GcpService.js

```diff
@@ -1,16 +1,39 @@
 const googleAuth = require('google-auto-auth');
+const async = require('async');
 const AWS = require('aws-sdk');
 const { errors } = require('arsenal');
 const Service = AWS.Service;
 
 const GcpSigner = require('./GcpSigner');
 
-function genAuth(authParams) {
-    const authOptions = authParams || {};
-    Object.assign(authOptions, {
-        scopes: 'https://www.googleapis.com/auth/cloud-platform',
-    });
-    return googleAuth(authOptions);
+function genAuth(authParams, callback) {
+    async.waterfall([
+        function authKeyFile(next) {
+            const authOptions = {
+                scopes: authParams.scopes,
+                keyFilename: authParams.keyFilename,
+            };
+            const auth = googleAuth(authOptions);
+            auth.getAuthClient(err => {
+                if (!err) return next(null, auth);
+                return next(null, null);
+            });
+        },
+        function authCredentials(authKeyfile, next) {
+            const authOptions = {
+                scopes: authParams.scopes,
+                keyFilename: authParams.credentials,
+            };
+            const auth = googleAuth(authOptions);
+            auth.getAuthClient(err => {
+                if (!err) return next(null, authKeyfile, auth);
+                return next(null, authKeyfile, null);
+            });
+        },
+        (authKeyfile, authCredentials, next) => {
+            next(null, authKeyfile || authCredentials || null);
+        },
+    ], (err, auth) => callback(auth));
```

Comment:
Why are you ignoring `err` and sending `auth` as first argument?

Reply:
Cleaned up the err issue with async.tryEach